site/src/components/Popover/Popover.tsx

```diff
@@ -0,0 +1,163 @@
+import {
+  ReactElement,
+  ReactNode,
+  cloneElement,
+  createContext,
+  useContext,
+  useRef,
+  useState,
+} from "react";
+// This is used as base for the main Popover component
+// eslint-disable-next-line no-restricted-imports -- Read above
+import MuiPopover, {
+  type PopoverProps as MuiPopoverProps,
+} from "@mui/material/Popover";
+
+type TriggerMode = "hover" | "click";
+
+type TriggerRef = React.RefObject<HTMLElement>;
```

Comment:
Do you think it's worth genericizing this? Something like:
```ts
type TriggerRef <Ele extends HTMLELement = HTMLElement> = React.RefObject<Ele>;
```

Reply:
Hm... I don't think so. I think HTMLElement is generic enough. Do you have any use case in mind?

Reply:
Well, genericizing might be the wrong word, but basically, adding type parameters. We're not exporting `TriggerRef`, but we are exporting the hook that contains it, and I'm worried that in some cases, `HTMLElement` might cause type mismatches because it isn't specific enough, and someone might try to reuse the ref from the hook for some other logic

Reply:
I think it is okay for now, I would not think too much about all possible use cases right now but only on the use cases that we have. We can change it in the future if it is needed.